src/main/scala/com/target/data_validator/validator/ColumnSumCheck.scala

```diff
@@ -0,0 +1,134 @@
+package com.target.data_validator.validator
+
+import com.target.data_validator.{ValidatorCheckEvent, ValidatorError, VarSubstitution}
+import io.circe._
+import io.circe.generic.semiauto._
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
+import org.apache.spark.sql.catalyst.expressions.aggregate.Sum
+import org.apache.spark.sql.types._
+
+case class ColumnSumCheck(
+  column: String,
+  minValue: Option[Json] = None,
+  maxValue: Option[Json] = None,
+  inclusive: Option[Json] = None
+) extends ColumnBased(column, Sum(UnresolvedAttribute(column)).toAggregateExpression()) {
+
+  private val minOrMax: Either[String, Unit] = (minValue, maxValue) match {
+    case (None, None) => Left("'minValue' or 'maxValue' or both must be defined")
+    case _ => Right()
```

Comment:
This feels cleaner when testing emptiness:

```suggestion
  private val minOrMax: Either[String, Unit] = if (minValue.isEmpty && maxValue.isEmpty) {
      Left("'minValue' or 'maxValue' or both must be defined")
    } else {
      Right()
    }
```